test/comparators.go

```diff
@@ -0,0 +1,62 @@
+package test
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// ErrorTypeComparator defines an error type comparator for use with go-cmp
+// Example:
+//  opts := cmp.Options{
+//  	 test.ErrorTypeComparator,
+//  }
+//
+//  if !cmp.Equal(err, MyCustomError{}, opts) {
+//  	t.Fatal(e.String())
+//  }
+//
+var ErrorTypeComparator = cmp.Comparer(func(x error, y error) bool {
```

Comment:
I suggest naming this `ErrorTypeComparer` for consistency and somewhat more correct language.

Reply:
Good suggestion. Done.